litmus-portal/authentication/api/routes/user_router.go

```diff
@@ -0,0 +1,184 @@
+package routes
+
+import (
+	"litmus/litmus-portal/authentication/api/middleware"
+	"litmus/litmus-portal/authentication/api/presenter"
+	"litmus/litmus-portal/authentication/pkg/entities"
+	"litmus/litmus-portal/authentication/pkg/user"
+	"litmus/litmus-portal/authentication/pkg/utils"
+
+	"github.com/gin-gonic/gin"
+	log "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+//UserRouter creates all the required routes for user authentications purposes.
+func UserRouter(router *gin.Engine, service user.Service) {
+	router.POST("/login", loginUser(service))
+	router.Use(middleware.JwtMiddleware())
+	router.POST("/update/password", updatePassword(service))
+	router.POST("/reset/password", resetPassword(service))
+	router.POST("/create", createUser(service))
+	router.POST("/update/details", updateUser(service))
+	router.GET("/users", fetchUsers(service))
+}
+
+func createUser(service user.Service) gin.HandlerFunc {
```

Comment:
Handlers / controllers like this could have been placed in a separate file, this can be kept just for all the routes. @DarthBenro008

Reply:
I do not want to export these functions (handlers) and make sure it's available under the same package and to the API only. The code architecture is highly inspired by Clean Architecture. Maybe the package name `routes` is misleading and could be changed to `handlers` instead?

Reply:
@DarthBenro008 by clean arch. we define the handler as a service, but in this case you have created a service file which has all component based services. There are two options - 

* Either you can create a custom middleware for the services if you feel these functions cannot be a part of the service routines.

*  If they can be then i would suggest to move this logic to the services itself and make them proper handlers.

Any approach can be taken but routing package should not have the handler logic or the middleware logic. It can be kept as a simple router with mapping of routes and their corresponding handlers.

Reply:
Changing the package name to handlers will be confusing for the people looking for the routing map in future.

Reply:
@ishangupta-ds given the options, I think I will take the Controller approach (i.e an API oriented package rather than a service-oriented package as those functions as Gin Handlers). I will call these "controllers" as "handlers" in the codebase as they as Gin Handler funcs after all. I do not want to make these service middlewares as they depend on the API that handles the request.

An example of UserRouter definition after the change:

```go
func UserRouter(router *gin.Engine, service user.Service) {
	router.POST("/login", handlers.LoginUser(service))
	router.Use(middleware.JwtMiddleware())
	router.POST("/update/password", handlers.UpdatePassword(service))
	router.POST("/reset/password", handlers.ResetPassword(service))
	router.POST("/create", handlers.CreateUser(service))
	router.POST("/update/details", handlers.UpdateUser(service))
	router.GET("/users", handlers.FetchUsers(service))
}
```

Reply:
Sure !